src/uct/sm/scopy/cma/cma_iface.c

```diff
@@ -145,6 +147,19 @@ static UCS_CLASS_DEFINE_NEW_FUNC(uct_cma_iface_t, uct_iface_t, uct_md_h,
                                  const uct_iface_config_t *);
 static UCS_CLASS_DEFINE_DELETE_FUNC(uct_cma_iface_t, uct_iface_t);
 
-UCT_TL_DEFINE(&uct_cma_component, cma, uct_sm_base_query_tl_devices,
-              uct_cma_iface_t, "CMA_", uct_cma_iface_config_table,
-              uct_cma_iface_config_t);
+void UCS_F_CTOR uct_cma_init(void)
+{
+    uct_component_register(cma);
+    uct_tl_register(cma);
+}
+
+void UCS_F_DTOR uct_cma_cleanup(void)
+{
+    uct_tl_unregister(cma);
+    uct_component_unregister(cma);
+}
```

Comment:
seems these lines are repeated in multiple places (cma, knem, posix, sysv), can we unite them to the TL_REGISTER macro somehow?
maybe UCT_TL_REGISTER_DECL macro would also accept custome init and cleanup code blocks (for xpmem)?

Reply:
> seems these lines are repeated in multiple places (cma, knem, posix, sysv), can we unite them to the TL_REGISTER macro somehow?
> maybe UCT_TL_REGISTER_DECL macro would also accept custome init and cleanup code blocks (for xpmem)?

ok, added macro